src/measurements.jl

```diff
@@ -895,3 +895,77 @@ function _fiducial_WH(::Type{T}, d::Integer) where {T}
 end
 
 _fiducial_WH(d::Integer) = _fiducial_WH(ComplexF64, d)
+
+"""
+    state_discrimination_min_error(
+        ρ::Vector{<:AbstractMatrix{T}},
+        q::Vector{<:AbstractFloat} = fill(1/length(ρ), length(ρ));
+        verbose = false,
+        dualize = false,
+        solver = Hypatia.Optimizer{_solver_type(T)}
+    ) where {T, S<:AbstractFloat}
+
+Return a POVM ``{E_i}_{=1}^N`` and probability P such that if we observe ``E_i`` with average probability P we are in state ``ρ_i``
+
+# Arguments
+- ρ: list of state to be discriminated
+- q: (optional) list of probabilities associated with the state ρ, if not provided, uniform probability is assumed
+"""
+function state_discrimination_min_error(
+    ρ::Vector{<:AbstractMatrix{T}},
+    q::Vector{S} = fill(1/length(ρ),length(ρ));
+    verbose = false,
+    dualize = false,
+    solver = Hypatia.Optimizer{_solver_type(T)}) where {T,S<:AbstractFloat}
+
+    model = JuMP.GenericModel{_solver_type(T)}()    
+    N = length(ρ)
+    d = maximum(size(ρ[1]))
+
+    @assert length(q) == length(ρ) 
+    @assert isapprox(sum(q), 1.0; atol=eps(S))
+
+    E = [JuMP.@variable(model, [1:d, 1:d] in JuMP.HermitianPSDCone()) for i in 1:N]
+    JuMP.@constraint(model, E[N] == I - sum(E[1:N-1]))
+
+    JuMP.@objective(
+        model,
+        Max,
+        sum(q[i] * real(dot(ρ[i]',E[i])) for i in 1:N),
```

Comment:
```suggestion
        sum(q[i] * real(dot(ρ[i],E[i])) for i in 1:N),
```
No adjoint needed, the states are by assumption Hermitian.